### PR TITLE
Add support for `FROM <filename>`, `DELIMITER`, and `CSV HEADER` options for `COPY` command

### PR DESCRIPTION
**Pull request by poonai**
This PR adds support for delimiter and csv header options in copy command

Signed-off-by: password <rbalajis25@gmail.com>

### CODE REVIEW

File: src/ast/mod.rs [hunk 1]
```diff
@@ -670,6 +670,12 @@ pub enum Statement {
         columns: Vec<Ident>,
         /// VALUES a vector of values to be copied
         values: Vec<Option<String>>,
+        /// file name of the data to be copied from
+        filename: Option<Ident>,
+        /// delimiter character
+        delimiter: Option<Ident>,
+        /// CSV HEADER
```
Comment (alamb) [inline src/ast/mod.rs:677]: I didn't think `CSV HEADER` was part of the syntax. Postgres seems to only support `HEADER`:

https://www.postgresql.org/docs/13/sql-copy.html

I think CSV is used as part of the FORMAT option, like

```sql
COPY ... FORMAT CSV ...
```

**Review by alamb — COMMENTED** [review]
Thanks for the contribution @poonai . I had a question about `CSV HEADER` 

the lint CI failures should be resolved if you `merge` up to main

File: src/ast/mod.rs (hunk 1 above)
Comment (poonai) [reply src/ast/mod.rs:677]: compactability

If you scroll down, you'll find the compatibility section. There you can find CSV HEADER spec. 

Sure, I merge the main

**Review by alamb — APPROVED** [review]
Thanks again @poonai  for the contribution. If/when people want further support for COPY options, we can always add it as a follow on PR

**Pull request merged**